src/units/key_manager.rs

```diff
@@ -97,11 +99,53 @@ impl KeyManager {
 
                 log::info!("Running {cmd:?}");
 
-                if let Err(e) = cmd.output() {
+                let output = cmd.output().map_err(|e| {
                     error!("[KM]: Error creating keyset: {e}");
+                    Terminated
+                })?;
+                if !output.status.success() {
+                    error!("[KM]: Create command failed with: {}", output.status);
```

Comment:
Should log statements here note which zone is concerned?

Reply:
I'll add the zone name.